src/Graph.cc

```diff
@@ -121,56 +121,86 @@ bool sea_dsa::Node::hasAccessedType(unsigned o) const {
   if (isOffsetCollapsed())
     return false;
   Offset offset(*this, o);
-  return m_accessedTypes.count(offset.getNumericOffset()) > 0 &&
-         !m_accessedTypes.at(offset.getNumericOffset()).isEmpty();
+
+  auto it = m_accessedTypes.find(offset.getNumericOffset());
+  return it != m_accessedTypes.end() && !it->second.isEmpty();
 }
 
-void sea_dsa::Node::addAccessedType(unsigned o, llvm::Type *t) {
-  if (isOffsetCollapsed())
-    return;
-  Offset offset(*this, o);
-  growSize(offset, t);
-  if (isOffsetCollapsed())
-    return;
+void sea_dsa::Node::addAccessedType(unsigned off, llvm::Type *type) {
```

Comment:
I noticed addAccessedTypes was a hotspot in SMC and modified it to take ~30% fewer cycles